gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java

```diff
@@ -0,0 +1,583 @@
+          return payload.getResourceDescriptorsList();
+        }
+      };
+
+  private static String ALL_SCOPES[] = {
+    "https://www.googleapis.com/auth/logging.read",
+    "https://www.googleapis.com/auth/logging.write",
+    "https://www.googleapis.com/auth/logging.admin"
+  };
+
+  /**
+   * A PathTemplate representing the fully-qualified path to represent
+   * a log_name resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  private static final PathTemplate LOG_NAME_PATH_TEMPLATE =
+      PathTemplate.create("/projects/{project}/logs/{log}");
+
+  // ========
+  // Members
+  // ========
+
+  private final ManagedChannel channel;
+  private final ServiceApiSettings settings;
+
+  // ===============
+  // Factory Methods
+  // ===============
+
+  /**
+   * Constructs an instance of LoggingServiceV2Api with default settings.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static LoggingServiceV2Api create() throws IOException {
+    return create(new ServiceApiSettings());
+  }
+
+  /**
+   * Constructs an instance of LoggingServiceV2Api, using the given settings. The channels are created based
+   * on the settings passed in, or defaults for any settings that are not set.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static LoggingServiceV2Api create(ServiceApiSettings settings) throws IOException {
+    return new LoggingServiceV2Api(settings);
+  }
+
+  /**
+   * Constructs an instance of LoggingServiceV2Api, using the given settings. This is protected so that it
+   * easy to make a subclass, but otherwise, the static factory methods should be preferred.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  protected LoggingServiceV2Api(ServiceApiSettings settings) throws IOException {
+    ServiceApiSettings internalSettings = ApiUtils.populateSettings(settings,
+        SERVICE_ADDRESS, DEFAULT_SERVICE_PORT, ALL_SCOPES);
+    this.settings = internalSettings;
+    this.channel = internalSettings.getChannel();
+  }
+
+  // ==============================
+  // Resource Name Helper Functions
+  // ==============================
+
+  /**
+   * Creates a string containing the fully-qualified path to represent
+   * a log_name resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String createLogNamePath(String project, String log) {
+    return LOG_NAME_PATH_TEMPLATE.instantiate(
+        "project", project,"log", log);
+  }
+
+
+  /**
+   * Extracts the project from the given fully-qualified path which
+   * represents a logName resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String extractProjectFromLogNamePath(String logNamePath) {
+    return LOG_NAME_PATH_TEMPLATE.parse(logNamePath).get("project");
+  }
+
+  /**
+   * Extracts the log from the given fully-qualified path which
```

Comment:
logNamePath contains two parameters: project and log. This method extracts the parameter log from the path. I agree that this is confusing. Tracking in #474 .